hudi-client/hudi-client-common/src/test/java/org/apache/hudi/table/upgrade/TestEightToSevenDowngradeHandler.java

```diff
@@ -0,0 +1,110 @@
+import org.apache.hudi.metadata.HoodieTableMetadataUtil;
+import org.apache.hudi.storage.HoodieStorage;
+import org.apache.hudi.storage.StoragePath;
+import org.apache.hudi.storage.hadoop.HoodieHadoopStorage;
+
+import org.apache.hadoop.conf.Configuration;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.MockedStatic;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.hudi.common.table.HoodieTableConfig.TABLE_METADATA_PARTITIONS;
+import static org.apache.hudi.metadata.MetadataPartitionType.FUNCTIONAL_INDEX;
+import static org.apache.hudi.metadata.MetadataPartitionType.PARTITION_STATS;
+import static org.apache.hudi.metadata.MetadataPartitionType.SECONDARY_INDEX;
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class TestEightToSevenDowngradeHandler {
+  private static final List<String> SAMPLE_METADATA_PATHS = Arrays.asList(
+      FUNCTIONAL_INDEX.getPartitionPath(),
+      SECONDARY_INDEX.getPartitionPath(),
+      PARTITION_STATS.getPartitionPath(),
+      "metadata_path1",
+      "metadata_path2");
+
+  @Mock
+  HoodieTableMetaClient mdtMetaClient;
+  @Mock
+  HoodieEngineContext context;
+
+  @Test
+  void testDeleteMetadataPartition() {
+    try (MockedStatic<HoodieTableMetadataUtil> mockedMetadataUtils = mockStatic(HoodieTableMetadataUtil.class)) {
+      List<String> leftPartitionPaths =
+          EightToSevenDowngradeHandler.deleteMetadataPartition(context, mdtMetaClient, SAMPLE_METADATA_PATHS);
+
+      mockedMetadataUtils.verify(
+          () -> HoodieTableMetadataUtil.deleteMetadataTablePartition(
+              mdtMetaClient, context, FUNCTIONAL_INDEX.getPartitionPath(), true),
+          times(1));
+      mockedMetadataUtils.verify(
+          () -> HoodieTableMetadataUtil.deleteMetadataTablePartition(
+              mdtMetaClient, context, SECONDARY_INDEX.getPartitionPath(), true),
+          times(1));
+      mockedMetadataUtils.verify(
+          () -> HoodieTableMetadataUtil.deleteMetadataTablePartition(
+              mdtMetaClient, context, PARTITION_STATS.getPartitionPath(), true),
+          times(1));
+
+      assertArrayEquals(new String[]{"metadata_path1", "metadata_path2"}, leftPartitionPaths.toArray());
+    }
+  }
+
+  @Test
+  void testDowngradeMetadataPartitions() {
+    HoodieStorage hoodieStorage = new HoodieHadoopStorage("any_path", new Configuration(false));
+    StoragePath basePath = new StoragePath("file:///base_path/.hoodie/metadata/.hoodie");
```

Comment:
Can we create a temp path which gets cleaned after test, instead of hardcoded path? See for usage of `@TempDir` in some other tests.

Reply:
Use the path to create `HoodieStorage` instance as well.

Reply:
Done.